fem/fe_coll.hpp

```diff
@@ -1305,6 +1305,60 @@ class Local_FECollection : public FiniteElementCollection
    virtual int GetContType() const { return DISCONTINUOUS; }
 };
 
-}
+/** @brief Radial basis function collection.
+    Instead of using polynomials as a basis, these methods use
+    Gaussian-like functions centered at chosen points. Reproducing
+    kernels augment the radial basis functions so that the resultant
+    basis can exactly represent functions of a chosen polynomial degree. */
+class KernelFECollection : public FiniteElementCollection
+{
+private:
+   int maxDim;
+   char d_name[32];
+   ScalarFiniteElement *Tr_Elements[Geometry::NumGeom];
+   ScalarFiniteElement *L2_Elements[Geometry::NumGeom];
+   int *SegDofOrd[2]; // for rotating segment dofs in 1D
+   int *OtherDofOrd;
+   bool ValidGeomType(Geometry::Type GeomType) const;
+
+public:
+   /** @brief Create a radial basis function or reproducing kernel finite
+       element collection.
+       @param D           Reference space dimension
+       @param numPointsD  Number of points across the element in each dimension
+       @param rbfType     Type of radial basis function, from RBFType
+       @param order       Order of polynomial correction, >= 0 for RK or -1 for RBF
+       @param h           Shape parameter, approximately equal to the number of points in the support radius in one dimension, should generally be > order
```

Comment:
80 characters per line, could be split into multiple lines.

Reply:
Done.